app/javascript/components/FilterForm.jsx

```diff
@@ -0,0 +1,150 @@
+import { useState } from "react";
+
+export default function FilterForm({ tags, userLevels, languages, onSubmit }) {
+  const [form, setForm] = useState({
+    tags_name_eq: "",
+    duration_eq: "",
+    user_level_eq: "",
+    user_language_eq: "",
+    starts_after: "",
+    starts_before: ""
+  });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setForm({ ...form, [name]: value });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    const payload = {
+      ...form,
+      periods_start_at_gteq: form.starts_after ? `${form.starts_after}T00:00` : "",
```

Comment:
Maybe we can try with dates, and send form.starts_after - 1 day and form.starts_before + 1 day?(not a big deal but it might look better than this hardcoded hours)